File: LeetCode/LeetCode/Problems/28/Problem2849.cs
using System;

public class Problem2849
{
    public bool IsReachableAtTime(int sx, int sy, int fx, int fy, int t)
    {
        var x_diff = Math.Abs(sx - fx);
        var y_diff = Math.Abs(sy - fy);
        return x_diff == 0 && y_diff == 0 ? t != 1 : Math.Max(x_diff, y_diff) <= t;
    }
}

File: leetcode_rust/src/problems/problem_2849.rs
pub fn is_reachable_at_time(sx: i32, sy: i32, fx: i32, fy: i32, t: i32) -> bool {
    let x_diff = sx.abs_diff(fx);
    let y_diff = sy.abs_diff(fy);

    if x_diff == 0 && y_diff == 0 {
        t != 1
    } else {
        x_diff.max(y_diff) <= t as u32
    }
}

pub fn is_reachable_at_time_leetcode_ver(sx: i32, sy: i32, fx: i32, fy: i32, t: i32) -> bool {
    let x_diff = (sx - fx).abs();
    let y_diff = (sy - fy).abs();

    if x_diff == 0 && y_diff == 0 {
        t != 1
    } else {
        x_diff.max(y_diff) <= t
    }
}
